### PR TITLE
Make DeviceLostClosure.from_c mutate the closure before dropping it.

### PR DESCRIPTION
**Pull request by bradwerth**
**Description**
Since `DeviceLostClosureC` requires that it be called before it is dropped, and we receive an existing `DeviceLostClosureC` struct as param in `DeviceLostClosure.from_c`, we need to mark the param as called before it drops out of scope.

**Testing**
Tested by embedders using `DeviceLostClosure.from_c`.

**Checklist**

- [X] Run `cargo fmt`.
- [X] Run `cargo clippy`. If applicable, add:
  - [ ] `--target wasm32-unknown-unknown`
  - [ ] `--target wasm32-unknown-emscripten`
- [X] Run `cargo xtask test` to run tests.
- [x] Add change to `CHANGELOG.md`. See simple instructions inside file.

### CODE REVIEW

File: wgpu-core/src/device/mod.rs [hunk 1]
```diff
@@ -294,14 +294,18 @@ impl DeviceLostClosure {
...
         // Build an inner with the values from closure, ensuring that
         // inner.called is false.
         let inner = DeviceLostClosureC {
             callback: closure.callback,
             user_data: closure.user_data,
             called: false,
         };
+
+        // Mark the original closure as called, so we can safely drop it.
+        closure.called = true;
```
Comment (cwfitzgerald) [inline wgpu-core/src/device/mod.rs]: This doesn't actually do anything - `closure` is dropped immediately after this.

**Review by cwfitzgerald — DISMISSED** [review]

File: wgpu-core/src/device/mod.rs (hunk 1 above)
Comment (nical) [reply wgpu-core/src/device/mod.rs]: It prevents the `drop` impl from panicking (it asserts that we don't leak the closure by checking `called`).

**Review by nical — APPROVED** [review]
nit: renaming `called` into `consumed` would make this easier to understand.

File: wgpu-core/src/device/mod.rs (hunk 1 above)
Comment (bradwerth) [reply wgpu-core/src/device/mod.rs]: I've changed the name of the member from "called" to "consumed" to clarify its purpose, and why it is mutated in this case.

**Pull request merged**